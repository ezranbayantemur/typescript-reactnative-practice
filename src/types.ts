export interface User {
  id: number;
  uid: string;
  password: string;
  first_name: string;
  last_name: string;
  username: string;
  email: string;
  avatar: string;
  gender: string;
  phone_number: string;
  social_insurance_number: string;
  date_of_birth: string;
  employment: Employment;
  address: Address;
  credit_card: Creditcard;
  subscription: Subscription;
}

interface Subscription {
  plan: string;
  status: string;
  payment_method: string;
  term: string;
}

interface Creditcard {
  cc_number: string;
}

interface Address {
  city: string;
  street_name: string;
  street_address: string;
  zip_code: string;
  state: string;
  country: string;
  coordinates: Coordinates;
}

interface Coordinates {
  lat: number;
  lng: number;
}

interface Employment {
  title: string;
  key_skill: string;
}
