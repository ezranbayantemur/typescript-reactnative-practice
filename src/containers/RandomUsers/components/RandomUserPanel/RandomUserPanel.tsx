import React, {FC} from 'react';
import {FlatList, View} from 'react-native';
import Button from '../../../../components/Button';
import UserItem from '../../../../components/list-item/UserItem';
import {User} from '../../../../types';

interface RandomUserPanelProps {
  data: Partial<User>[];
  loading: boolean;
  onClear: () => void;
  onRefresh: () => void;
}

const RandomUserPanel: FC<RandomUserPanelProps> = ({
  data,
  loading,
  onClear,
  onRefresh,
}) => {
  return (
    // eslint-disable-next-line react-native/no-inline-styles
    <View style={{flex: 1}}>
      <FlatList data={data} renderItem={({item}) => <UserItem user={item} />} />
      <Button title="Temizle" onPress={onClear} type="outline" />
      <Button title="Yenile" onPress={onRefresh} loading={loading} />
    </View>
  );
};

export default RandomUserPanel;
