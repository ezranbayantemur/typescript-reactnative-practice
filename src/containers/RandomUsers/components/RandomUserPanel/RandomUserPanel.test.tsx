import {render} from '@testing-library/react-native';
import React from 'react';
import {User} from '../../../../types';
import RandomUserPanel from './RandomUserPanel';

const noop = jest.fn();

test('should render correctly', () => {
  const comp = render(
    <RandomUserPanel data={[]} loading onRefresh={noop} onClear={noop} />,
  );

  expect(comp).toMatchSnapshot();
});

test('should render data correctly', () => {
  const mockData: Partial<User>[] = [
    {avatar: '', username: '', first_name: '', last_name: ''},
    {avatar: '', username: '', first_name: '', last_name: ''},
  ];

  const comp = render(
    <RandomUserPanel data={mockData} loading onRefresh={noop} onClear={noop} />,
  );

  const userItems = comp.queryAllByTestId('user-item');
  expect(userItems.length).toBe(mockData.length);
});
