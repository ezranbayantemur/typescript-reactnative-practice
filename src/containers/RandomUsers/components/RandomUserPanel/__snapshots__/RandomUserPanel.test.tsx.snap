// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render correctly 1`] = `
<View
  style={
    Object {
      "flex": 1,
    }
  }
>
  <RCTScrollView
    data={Array []}
    disableVirtualization={false}
    getItem={[Function]}
    getItemCount={[Function]}
    horizontal={false}
    initialNumToRender={10}
    keyExtractor={[Function]}
    maxToRenderPerBatch={10}
    onContentSizeChange={[Function]}
    onEndReachedThreshold={2}
    onLayout={[Function]}
    onMomentumScrollEnd={[Function]}
    onScroll={[Function]}
    onScrollBeginDrag={[Function]}
    onScrollEndDrag={[Function]}
    removeClippedSubviews={false}
    renderItem={[Function]}
    scrollEventThrottle={50}
    stickyHeaderIndices={Array []}
    updateCellsBatchingPeriod={50}
    viewabilityConfigCallbackPairs={Array []}
    windowSize={21}
  >
    <View />
  </RCTScrollView>
  <View
    accessible={true}
    focusable={true}
    onClick={[Function]}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      Object {
        "alignItems": "center",
        "borderColor": "#8e24aa",
        "borderRadius": 5,
        "borderWidth": 1,
        "justifyContent": "center",
        "margin": 10,
        "opacity": 1,
        "padding": 10,
      }
    }
    testID="button-touchable"
  >
    <Text
      style={
        Object {
          "color": "#8e24aa",
          "fontSize": 15,
        }
      }
      testID="button-text"
    >
      Temizle
    </Text>
  </View>
  <View
    accessible={true}
    focusable={true}
    onClick={[Function]}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      Object {
        "alignItems": "center",
        "backgroundColor": "#8e24aa",
        "borderRadius": 5,
        "justifyContent": "center",
        "margin": 10,
        "opacity": 1,
        "padding": 10,
      }
    }
    testID="button-touchable"
  >
    <ActivityIndicator
      animating={true}
      color="#999999"
      hidesWhenStopped={true}
      size="small"
      testID="button-indicator"
    />
  </View>
</View>
`;
